test/e2e/storage/persistent_volumes-disruptive.go

```diff
@@ -256,36 +256,92 @@ func tearDownTestCase(c clientset.Interface, f *framework.Framework, ns string,
 // - If `service` also returns stderr "command not found", the test is aborted.
 // Allowed kubeletOps are `kStart`, `kStop`, and `kRestart`
 func kubeletCommand(kOp kubeletOpt, c clientset.Interface, pod *v1.Pod) {
+	command := ""
+	sudoPresent := false
+	systemctlPresent := false
+	kubeletPid := ""
+
 	nodeIP, err := framework.GetHostExternalAddress(c, pod)
 	Expect(err).NotTo(HaveOccurred())
 	nodeIP = nodeIP + ":22"
-	systemctlCmd := fmt.Sprintf("sudo systemctl %s kubelet", string(kOp))
-	framework.Logf("Attempting `%s`", systemctlCmd)
-	sshResult, err := framework.SSH(systemctlCmd, nodeIP, framework.TestContext.Provider)
+
+	framework.Logf("Checking if sudo command is present")
```

Comment:
why do you need to check sudo? ssh should use root id.

Reply:
Framework is flexible and allows user to ssh with non root user account. We can set the user account to login with in the environment variable as below.

```export KUBE_SSH_USER="user"```

For non-root user account we need sudo. 

In our setup we do not have sudo package installed, and we are using root user to login. checking existence of sudo command, helps us prepare right command for restarting kubelet.

Reply:
How is kubelet restart done in your test if no sudo installed? Test needs sudo to restart kubelet [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/persistent_volumes-disruptive.go#L262) and [in gce](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L3803)

Reply:
That's the change we have made in this PR. we do not need sudo prefixed in the systemctl command we issue on our node.

We first check if sudo command is present or not.
```
framework.Logf("Checking if sudo command is present")
sshResult, err := framework.SSH("sudo --version", nodeIP, framework.TestContext.Provider)
	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("SSH to Node %q errored.", pod.Spec.NodeName))
if !strings.Contains(sshResult.Stderr, "command not found") {
		sudoPresent = true
	}
```

If sudo command is not present we will not include it in the command for restarting kubelet.

```
if sudoPresent {
     command = fmt.Sprintf("sudo %s", command)
}
```

Here command is either 

```
command = fmt.Sprintf("systemctl %s kubelet", string(kOp))
```
or 
```
command = fmt.Sprintf("service kubelet %s", string(kOp))
```


Change is safe for GCE.

Reply:
ok, thanks for details.